vllm/utils.py

```diff
@@ -2925,6 +2925,44 @@ def is_torch_equal_or_newer(target: str) -> bool:
         return Version(importlib.metadata.version('torch')) >= Version(target)
 
 
+def build_uri(scheme: str,
+              host: str,
+              port: Optional[int] = None,
+              path: str = "",
+              params: str = "",
+              query: str = "",
+              fragment: str = "") -> str:
+    """
+    Robustly build a URI that properly handles IPv6 addresses.
+    
+    Args:
+        scheme: URI scheme (e.g., 'http', 'https')
+        host: hostname or IP address
+        port: port number (optional)
+        path: path component
+        params: parameters component
+        query: query string
+        fragment: fragment identifier
+    
+    Returns:
+        Complete URI string
+    """
+    # Handle IPv6 addresses
+    if host:
+        try:
+            # Check if it's an IPv6 address
+            ip = ipaddress.ip_address(host)
+            # Ensure IPv6 addresses are bracketed
+            if (isinstance(ip, ipaddress.IPv6Address)
+                    and not (host.startswith('[') and host.endswith(']'))):
+                host = f'[{host}]'
```

Comment:
You can use this util for checking for a v6 address

https://github.com/vllm-project/vllm/blob/9ff2af6d2ba1757c6e59fe803225a73e61fe526f/vllm/utils/__init__.py#L815-L820